packages/integrations/components/integrationsManager/index.js

```diff
@@ -0,0 +1,144 @@
+  integrations,
+  isHydrated,
+  hydrateComponents,
+  hydratedComponents,
+}) => {
+  // Declare a map of available integration components with a corresponding key.
+  const componentMap = [
+    { key: 'googleAnalytics', el: GoogleAnalytics },
+  ];
+  // This check is run on server-side renders to ensure that the component hydration
+  // is ran a single time. After hydration, `isHydrated` is set to `true` and will
+  // only be re-run if the component props change during the component's lifecycle.
+  if (false === isHydrated) {
+    // Retrieve the keys available for render in the component map.
+    const keyMap = Object.keys(integrations);
+
+    // Create an array for hydrated component keys and props.
+    const components = [];
+
+    componentMap.forEach((component, index) => {
+      // Get the component's key.
+      const { key } = component;
+
+      if (- 1 < keyMap.indexOf(key)) {
+        // Retrieve the key/value pairs set for the integration and convert them into props.
+        // If a user config override exists, use it instead.
+        let props;
+        if ('integrations' in userConfig && key in userConfig.integrations) {
+          props = userConfig.integrations[key];
+        } else {
+          props = integrations[key];
+        }
+        // Update the item at the current index with the transformed component.
+        components[index] = { key, props };
+      }
+    });
+
+    // Update the components in the store.
+    hydrateComponents(components);
+  }
+
+  /**
+   * A function that takes a component key/props pair, matches the key with a
+   * component that exists in the `componentMap`, and returns a valid React
+   * element with the associated key and props applied.
+   *
+   * @param {Object} component - The component key and props.
+   */
+  const renderComponents = (component) => {
+    // Create a map of available component keys.
+    const keyMap = componentMap.map((co) => co.key);
+    const index = keyMap.indexOf(component.key);
+
+    // If the current component key exists in the key map, clone that component
+    // with the hydrated props for render.
+    if (- 1 < index) {
+      const props = { key: component.key, ...component.props };
+      // Clone the element.
+      return React.createElement(componentMap[index].el, props, null);
+    }
+
+    return null;
+  };
+
+  if (0 < hydratedComponents.length) {
+    return (
+      <>
+        {hydratedComponents.map(renderComponents)}
+      </>
+    );
+  }
+
+  return null;
+};
+
+IntegrationsManager.defaultProps = {
+  integrations: {},
+  hydratedComponents: [],
+};
+
+IntegrationsManager.propTypes = {
+  integrations: PropTypes.shape({
+    googleAnalytics: PropTypes.shape({
+      trackingId: PropTypes.string,
+    }),
+  }),
+  isHydrated: PropTypes.bool.isRequired,
+  hydrateComponents: PropTypes.func.isRequired,
+  hydratedComponents: PropTypes.arrayOf(PropTypes.object),
+};
+
+const mapStateToProps = (state, ownProps) => {
```

Comment:
I've been unofficially leaning toward using `react-redux` `useSelector` and `useDispatch` hooks instead of the connect functions for a couple reasons:
* A bit more concise than connect functions
* Makes it clearer what values are coming from state and what from API or user

Open to differing opinions, though, if you have some!

Reply:
I definitely think that the hooks are useful in a lot of applications, but IMO I think the connect functions are the better configuration in this instance. This is mainly because the selector hook can return any value, not just an object, and in this case I think we want to be pretty strict about the type safety of our props. Additionally we don't have access to `ownProps` when using the selector hook, so extra care would have to be taken there to get the same functionality. I don't think there are any detriments to the dispatch hook, but since the connect pattern is already present for `mapStateToProps`, I think it's consistent to use the `mapDispatch` function here too.

Reply:
Makes sense to me!